pkg/status/status-controller.go

```diff
@@ -374,14 +377,36 @@ func (c *Controller) runWorkStatusInformer(ctx context.Context) {
 	<-ctx.Done()
 }
 
-func shouldSkipUpdate(old, new interface{}) bool {
+func shouldSkipAdd(wdsName string, obj interface{}) bool {
+	if objNotInThisWDS(obj, wdsName) {
+		return true
+	}
+	return false
+}
+
+func shouldSkipUpdate(wdsName string, old, new interface{}) bool {
```

Comment:
This is not really  shouldSkipUpdate()  as it performs the actual update in case of false. Maybe separate and have a general skip as I suggested and add a function for update() that is conditioned by the shouldSKip ?

Reply:
Also, I'm trying to follow: I don't see actions performed in `shouldSkipUpdate()`, so what actions are we discussing about?

Reply:
@waltforme You are correct. My bad.